pkg/daemon/daemon.go

```diff
@@ -1721,12 +1658,31 @@ func (dn *Daemon) prepUpdateFromCluster() (*mcfgv1.MachineConfig, *mcfgv1.Machin
 // "transient state" file, which signifies that all of those prior steps have
 // been completed.
 func (dn *Daemon) completeUpdate(desiredConfigName string) error {
-	if err := dn.cordonOrUncordonNode(false); err != nil {
-		return err
+	if err := dn.nodeWriter.SetDesiredDrainer(fmt.Sprintf("%s-%s", "uncordon", desiredConfigName)); err != nil {
+		return fmt.Errorf("Could not set drain annotation: %w", err)
+	}
+
+	if err := wait.Poll(10*time.Second, 10*time.Minute, func() (bool, error) {
+		node, getErr := dn.kubeClient.CoreV1().Nodes().Get(context.TODO(), dn.name, metav1.GetOptions{})
+		if getErr != nil {
+			return false, fmt.Errorf("Could not poll node status. Assuming something went wrong")
```

Comment:
In general, we really should avoid turning transient network blips into any kind of fatal failure.  OK looking through things I think this will not be actually fatal, we should retry.

But it's not good for us to swallow the error; will do a quick patch.